### ISSUE TITLE
does boxr need a logo/hex-sticker?

### ISSUE THREAD

**Issue by nathancday** [r-box/boxr#92]
If boxr is getting a fresh Pkgdown site, I feel like it really needs a hex logo to go with it 😎. Thoughts? Design contest?

**Comment by ijlyttle** [comment]
I think a hex-logo would be great! 

I have to confess that I am obsessed with getting things automated at the moment.

The only idea I have would be to use "box" blue, then to use the "box" font and add an "r" in that style (which may be tricky because I don't think it's a real font and is likely trademarked). 

![image](https://user-images.githubusercontent.com/2800934/53293012-a89a9780-3791-11e9-9fb6-4f27b723b923.png)

I'm sure you have better ideas!

**Comment by nathancday** [comment]
First pass using the R language R. 

![logo](https://user-images.githubusercontent.com/808225/53294699-a5b39d00-37b9-11e9-91cc-41add034e8c1.png)

The Box logo is trademarked and they explicitly prohibit use, but I found a google font [Comfortaa](https://fonts.google.com/specimen/Comfortaa) that is very similar. There are several Box "blues" floating around I grabbed the hex #0061d5 from [this official logo](https://cloud.app.box.com/s/v1yn0eyqpxx657brrgcn/file/8677792621) Check out the updated README on my branch [logo](https://github.com/nathancday/boxr/tree/logo)

**Comment by nathancday** [comment]
Made another...

![boxr_logo_dark](https://user-images.githubusercontent.com/808225/53299738-927ded00-380c-11e9-8d7d-6e93e1826e5c.png)

cloud icon: https://www.flaticon.com/free-icon/cloud_1163624
icon author: https://www.flaticon.com/authors/iconixar

**Comment by ijlyttle** [comment]
I like them, especially the cloud! One thing I wonder about is the uppercase "R". I see what you are doing, using the R logo, but - as you know - package names are case-sensitive. 

I don't know which of these concerns is more important, but I thought I might bring it up.

As a crazy idea, could we try a lower-case "r", using the Comfortaa font, using the "R"-logo blue?

I also wonder if we should get a domain for the pkgdown website, perhaps `boxr.rbind.io`? That way, we could put the URL on the sticker. At some point, we should get @brendan-r's input on that.

**Comment by nathancday** [comment]
Here is the lower case r you suggested, I like it a lot.
![boxr_logo_dark2](https://user-images.githubusercontent.com/808225/53302774-18f7f600-3830-11e9-9578-491e2fbbf341.png)

I think we should definitely include a URL on the final sticker, either to a new rbind.io site or at least to the github repo.

**Comment by ijlyttle** [comment]
This looks great! From my point-of-view, the "final" sticker should look very similar.

From my point-of-view, just a couple of small suggestions: could the letters be moved closer together (but not so close that the Box folks would have a problem with it), and the horizontal alignment of the lettering be centered? I think the "b" is in a great place, but could the letters be moved closer together such that the space from the left-edge to the "b" matches the space from the right edge to the "r".

![image](https://user-images.githubusercontent.com/2800934/53303351-f7e2d580-382e-11e9-8776-a33028f4c926.png)

I hacked this from yours - and I don't claim that this spacing is ideal (just a first pass) - as well, I imagine the tool you are using would generate an svg?

To summarize - I really like this!

**Comment by nathancday** [comment]
![boxr_dark2 4](https://user-images.githubusercontent.com/808225/53305870-84ec5580-3854-11e9-9e01-dec48f38ea2e.png)

Is this getting closer? I've been staring at it for too long and I can't tell if the spacing is even or not 😜 

Yes I can export to SVG but Github doesn't like it, FYI using https://vectr.com/ for the fine-tuning

**Comment by ijlyttle** [comment]
I like it! 

I don't know if the spacing might be a little too close that Box might object - perhaps if show a little daylight between the "b" and the "o", and then a little more room between the "o" and the "x"

At the risk of being *far* to specific: how about if we moved:

- cloud 2 px left, balancing the space on either side to the edge
- "b" 5 px left, balancing the left of the "b" with the right of the "r"
- "o" 2 (or 3) px left, creating some space between the "b" and the "o" (to differentiate from Box)
- leave "x" and "r" where they are

Just an idea... but I think we are really close!

**Comment by ijlyttle** [comment]
Hi @brendan-r and @nathancday - another crazy idea here - we could go "legit", and apply to be a part of Box' Technology Partner program:

https://www.box.com/about-us/faq
https://community.box.com/t5/Getting-Started-for-Box-Partners/ct-p/BoxPartners
https://cloud.app.box.com/v/technologypartnerprogramguide

I suspect that we would need to maintain editorial control of "our" stuff, but it could help us use the Box logo with their blessing and possibly even get us some support, e.g. a legit burner account. I mean, we *are* making their lives easier, in addition to our own...

What do you think?

**Comment by nathancday** [comment]
Updated logo:

![logo2 4](https://user-images.githubusercontent.com/808225/53684281-54d00700-3cd9-11e9-8c86-b9e69f9f81ca.png)

I agree this package make Box a lot more attractive for R users, but I'm not sure it fits the "partner" qualifier. Quickly reading those links it seems like its aimed at for profit software products that are wrapping Box features in their interface.

**Comment by nathancday** [comment]
@ijlyttle Adding this here because it is were we discussed formal Box support before

Recently Box account support reached out to my team at @hemoshear because of our existing enterprise account, so I took the initiative and asked a few of our recent boxr focused questions.

Here are there responses:

"Thanks for sharing the information, always exciting to meet community developers building on Box.

From ongoing development perspective and having dedicated test accounts, this is actually something that you would be able to create yourself. If you sign up for a new account through https://developers.box.com you would be granted with a developer enterprise, with access to the admin console and ability to invite other developers / maintainers into the enterprise for testing purposes. This type of account is actually designed for testing and CI, and would give you essentially a sandbox enterprise for you to be able to work on the project without worries of touching any live data.

On the same note of continuing developing this project and alternate authentication models, we would suggest taking a look at our JWT Authentication model which works much better for solutions with no browser access. Initial documentation can be found here https://developer.box.com/docs/setting-up-a-jwt-app it's a decently simple approach, at it's core just being a public/private key exchange. Some Box built solutions utilize this approach exclusively for similar reasons as you stated - currently our CLI is able to read the JSON configuration file that can be generated from the developer console, and then generate access tokens in a programmatic fashion without the need to ever log in / touch the browser itself. The Box CLI can be found here for some quick reference as well - https://github.com/box/boxcli

From the project endorsement side, we currently already have boxr in our developer website as a Community Project - https://developer.box.com/docs/community-supported-projects - but unfortunately we currently do not have any structured way for our security, engineering and compliance teams to review these projects on an ongoing basis to be able to endorse any of them in an official capacity."

I haven't explored the JWT auth route yet, but it seems promising and R has [support already](https://cran.r-project.org/web/packages/jose/vignettes/jwt.html). I would say we add JWT to the RoadMap, becasue it has the potential to impact not only new testing related issues like #76 but also long standing one like #23.

**Comment by ijlyttle** [comment]
Hi @nathancday,

That's great news and I'm enthused about all your suggestions. I had no idea we were on the Box community-development page.

I can completely understand them not being in a position to endorse us - but I wonder if that precludes us from using their logo on a hex-sticker without getting into trouble. Is that worth asking, or do you think they have already answered?

I had tried to wrap my head around the JWT Authentication model a little while back, but I could not get anywhere. However, that does **not** mean we should not try. I was not aware of the "jose" package, that would seem to clear the way, and I fully trust Jeroen's work!

Let's put this on the roadmap!

I have some ideas about how we can organize ourselves, which touches on the developer account, but I will want to consult @brendan-r, to make sure he's on board.

Thanks!

**Comment by nathancday** [comment]
I had no idea we were on the community listing either. Nice surprise!

I say we use the logo until they ask us to stop, I doubt they ever will.

I'm strapped for bandwidth over the next week but after that I'm interested in diving into the JWT token idea via `jose` and seeing what can be done for server solutions. If we can figure something out better than auth locally >> scp >> pray method that would be huge.

And the testing revamp goes hand in hand with making a new boxr-test-dev account, so I am interested to help out on that too as we go after v0.3.6. Looks like we are still in que behind `rio` for v0.3.5....

**Comment by brendan-r** [comment]
That hex-logo design is extremely cool!

> I say we use the logo until they ask us to stop, I doubt they ever will.

This seems reasonable to me

**Comment by ijlyttle** [comment]
I agree! I suggest we move forward with [this one](https://github.com/r-box/boxr/issues/92#issuecomment-468933424) (which I somehow missed until now, sorry @nathancday), and fine-tune it as need be.

**Comment by ijlyttle** [comment]
I'll leave this issue open for now, just because we may have a little (low-priority) fine-tuning to do:

- pkgdown recommends that the size of the png be 240 x 278
- we may want to use the box logo itself, with the r added

I think it looks great now: https://r-box.github.io/boxr/dev/

Thanks @nathancday!

**Comment by nathancday** [comment]
Thanks for all of the work you have been doing migrating the repo!

Here is a link to the [logo project](https://vectr.com/nathancday/cyqTxHCTr) on Vectr, so you should be able to edit (I think). Note that it works best in Chrome.

Also here is a larger PNG in the pkgdwn dims.
![boxr_logo(1)](https://user-images.githubusercontent.com/808225/55279497-6c59da00-52ef-11e9-9d57-53ef830160dc.png)

FWIW All of the current components are open (cloud/font), so we aren't using anything that's trademarked. We could tighten letter spacing to make the "b" and "o" touch, which would be almost identical to the current official logo.
